clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/BackgroundEventHandler.java

```diff
@@ -30,9 +32,24 @@
 public class BackgroundEventHandler {
 
     private final Queue<BackgroundEvent> backgroundEventQueue;
+    private final KafkaConsumerMetrics kafkaConsumerMetrics;
+    private final KafkaShareConsumerMetrics kafkaShareConsumerMetrics;
+    private boolean isShareConsumer;
 
-    public BackgroundEventHandler(final Queue<BackgroundEvent> backgroundEventQueue) {
+    public BackgroundEventHandler(final Queue<BackgroundEvent> backgroundEventQueue, KafkaConsumerMetrics kafkaConsumerMetrics) {
         this.backgroundEventQueue = backgroundEventQueue;
+        this.kafkaConsumerMetrics = kafkaConsumerMetrics;
+        this.kafkaShareConsumerMetrics = null;
+        isShareConsumer = false;
```

Comment:
you don't really need this - option one: have kafkaConsumerMetrics and shareConsumerMetrics to impelment a common interface and do if(metrics instanceOf [one of the metrics class]) 

option two: either do null check or use optional.  Optional is better imo.